modules/decima-ui/src/main/java/com/shade/decima/ui/editor/core/menu/FollowReferenceItem.java

```diff
@@ -33,7 +33,19 @@ public class FollowReferenceItem extends MenuItem {
     @Override
     public void perform(@NotNull MenuItemContext ctx) {
         final Editor editor = ctx.getData(PlatformDataKeys.EDITOR_KEY);
-        final RTTIReference reference = (RTTIReference) ((CoreNodeObject) ctx.getData(PlatformDataKeys.SELECTION_KEY)).getValue();
+        final Object value = ((CoreNodeObject) ctx.getData(PlatformDataKeys.SELECTION_KEY)).getValue();
+        final RTTIReference reference;
+        if (value instanceof RTTIReference ref) {
+            reference = ref;
+        } else if (value instanceof RTTIObject obj){
+            if (obj.type().getTypeName().equals("TextureBindingWithHandle")) {
+                reference = obj.ref("TextureResource");
+            } else {
+                reference = obj.ref("Texture");
+            }
+        } else {
+            reference = null;
```

Comment:
1. The `findNode` method expects `reference` to be not null. I assume it can't be null in this case, but it would be better to assert that it won't even reach that `else` branch.
2. Can you please extract this logic of retrieving a reference from both `perform` and `isVisible` methods and reuse it in these places?

Reply:
Maybe the follow reference menu item should be extend to a more generell solution, like you already posted on discord (see picture), so that a list of references found in the sub tree of the selected node will be shown to choose.
![grafik](https://github.com/ShadelessFox/decima/assets/49768797/0c3589da-907e-42f2-ab4f-66006f47b669)

Reply:
I prefer a more general solution, yes. I can finish and commit it later today.